packages/calcite-components/src/components/button/button.scss

```diff
@@ -1,64 +1,90 @@
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-button-background-color: Specifies the component's background color when appearance="solid" or appearance="outline-fill".
+ * @prop --calcite-button-border-color: Specifies the component's border color when it has appearance="outline" or appearance="outline-fill".
+ * @prop --calcite-button-corner-radius: Specifies the component's corner radius.
+ * @prop --calcite-button-loader-color: Specifies the component's loader color.
+ * @prop --calcite-button-shadow-color: Specifies the component's box-shadow color.
+ * @prop --calcite-button-text-color: Specifies the component's text color.
+ * @prop --calcite-button-border-width: Specifies the component's border width.
```

Comment:
For @driskull case - would this need to be inline-border-start-width / inline-border-end-width? 

I'd almost suggest marking those as internal as well - we may be able to support the "button group" use case in a different way down the line.